oshdb-api/src/main/java/org/heigit/bigspatialdata/oshdb/api/mapreducer/GeometrySplitter.java

```diff
@@ -123,7 +141,7 @@
    * @param data the OSMContribution to split into the given sub-regions
    * @return a list of OSMContribution objects
    */
-  public List<Pair<U, OSMContribution>> splitOSMContribution(OSMContribution data) {
+  public List<IndexData<U,OSMContribution>> splitOSMContribution(OSMContribution data) {
```

Comment:
I _think_, we could implement this whole thing as a `Map<U, OSMContribution>` instead of a `List<IndexData<U,OSMContribution>>` (since the list will at most contain one entry per index value). :thinking:

Reply:
then we could get rid of the `IndexData` class altogether. In the MapReducer class, we would use this method then like this:

```java
    ret = this.flatMap(x -> gs.splitOSMContribution((OSMContribution) x).entrySet())
            .aggregateBy(Entry::getKey, geometries.keySet()).map(Entry::getValue);
```